service/codegen/src/main/java/io/helidon/service/codegen/ServiceDescriptorCodegen.java

```diff
@@ -1463,41 +1553,34 @@ private void providedTypeMethod(ClassModel.Builder classModel, DescribedService
                 ? service.providedDescriptor().typeName()
                 : service.serviceDescriptor().typeName();
 
-        classModel.addField(field -> field
-                .isStatic(true)
-                .isFinal(true)
-                .accessModifier(AccessModifier.PRIVATE)
-                .type(TypeNames.TYPE_NAME)
-                .name("PROVIDED_TYPE")
-                .addContentCreate(providedType));
+        String name = typeConstant(fieldHandler, providedType);
 
         // TypeName providedType()
         classModel.addMethod(method -> method.addAnnotation(Annotations.OVERRIDE)
                 .returnType(TypeNames.TYPE_NAME)
                 .name("providedType")
-                .addContentLine("return PROVIDED_TYPE;"));
+                .addContent("return ")
+                .addContent(name)
```

Comment:
Nitpick: Variable is called `constant` on line 1575 but `name` here, could be more readable following some convention